model/facade/action/GetPersonalSummaryByLoginDateAction.php

```diff
@@ -69,13 +78,69 @@ class GetPersonalSummaryByLoginDateAction extends Action{
      * @param DateTime $date the date on which we want to compute the summary.
      */
     public function __construct(UserVO $userVO, DateTime $date) {
+        $dao = DAOFactory::getJourneyHistoryDAO();
         $this->userVO = $userVO;
         $this->date = $date;
+        $this->currentJourney = $dao->getByIntervals( $this->date, $this->date, $this->userVO->getId())[0];
         $this->preActionParameter="GET_PERSONAL_SUMMARY_BY_USER_LOGIN_DATE_PREACTION";
         $this->postActionParameter="GET_PERSONAL_SUMMARY_BY_USER_LOGIN_DATE_POSTACTION";
 
     }
 
+    /**
+     * @return mixed
+     * @throws null
+     */
+    private function getWorkableHoursInThisJourneyPeriod() {
+        $initDate = $this->currentJourney->getInitDate();
+        $endDate = $this->currentJourney->getEndDate();
+
+        //Now need to find out the workable hours in this year
+        $extraHoursAction = new ExtraHoursReportAction($initDate, $endDate, $this->userVO);
+        $results = $extraHoursAction->execute();
+        return $results[1][$this->userVO->getLogin()]['workable_hours'];
+
+    }
+
+    /**
+     * @return float
+     */
+    private function getNumberOfWeeksInThisJourney() {
+        $initDate = $this->currentJourney->getInitDate();
+        $endDate = $this->currentJourney->getEndDate();
+
+        $interval = $initDate->diff( $endDate );
+        return ceil($interval->days/7);
+    }
+
+    /**
+     * @return mixed
+     * @throws null
+     */
+    private function getWorkedHoursInThisJourneyPeriod() {
+        $initDate = $this->currentJourney->getInitDate();
+        $endDate = $this->currentJourney->getEndDate();
+
+        // We need to get the number of hours worked till date
+        $extraHoursAction = new ExtraHoursReportAction($initDate , $endDate, $this->userVO);
+        $results = $extraHoursAction->execute();
+        return $results[1][$this->userVO->getLogin()]['total_hours'];
+    }
+
+    /**
+     * @return float
+     */
+    private function getWeeksTillEndOfJourneyPeriod() {
+        $lastDayOfJourney = $this->currentJourney->getEndDate();
+
+        $interval = $this->date->diff( $lastDayOfJourney );
+        return floor($interval->days/7);
```

Comment:
I'm getting a 0 weekly goal in the w52 example. If I change this `floor` with `ceil`, it works properly.